index.js

```diff
@@ -189,63 +192,58 @@ const displayChecks = () => {
   }
 };
 
-const displayRatioResult = () => {
-  let firstColor = foregroundColor.value;
-  let secondColor = backgroundColor.value;
+/**
+ * This function takes a value representing a color
+ * and returns an array of its RGB values.
+ * Accepted formats:
+ * - HEX format: color = "#FFF" | "#FFFFFF"
+ * - HSL format: color = "hsl(240, 100%, 50%)"
+ * - RGB format: color = rgb(100, 1, 233)
+ * - Named format: color = "white" | "chocolate"
+ *
+ * @param {String} color The color representation.
+ *
+ * @return {Array|null} RGB representation of the color eg. `[255, 0, 127]`
+ * or `null` when an incorrect format was provided or when color name
+ * could not be found.
+ */
+const getRGBfromColor = (color) => {
+  color = color.trim().toLowerCase();
+
+  if (hexRegex3Digit.test(color)) {
+    return hexToRGB(shortToFullHex(color));
+  }
+  
+  if (hexRegex.test(color)) {
+    return hexToRGB(color);
+  }
 
-  // CASE two Hexes
-  if (/^#.*/.test(firstColor) && /^#.*/.test(secondColor)) {
-    if (firstColor.length === 7 && secondColor.length === 7) {
-      if (hexRegex.test(firstColor) && hexRegex.test(secondColor)) {
-        ratioResult.innerHTML = colorFormatRatio(
-          firstColor,
-          secondColor,
-          hexToRGB
-        );
-      }
-    }
-    if (firstColor.length >= 7 && secondColor.length >= 7) {
-      if (!hexRegex.test(firstColor) || !hexRegex.test(secondColor)) {
-        showErrorMessage(warning);
-      }
-    } else if (firstColor.length < 7 || secondColor.length < 7) {
-      ratioResult.innerHTML = "";
-    }
-    if (hexRegex3Digit.test(firstColor) && hexRegex3Digit.test(secondColor)) {
-      ratioResult.innerHTML = colorFormatRatio(
-        shortToFullHex(firstColor),
-        shortToFullHex(secondColor),
-        hexToRGB
-      );
-    }
-  } else if (isNotEmpty(firstColor) && isNotEmpty(secondColor)) {
-    // CASE two RGBs
-    if (isValidRGB(firstColor) && isValidRGB(secondColor)) {
-      ratioResult.innerHTML = colorFormatRatio(
-        firstColor,
-        secondColor,
-        rgbInputToRGBNumbers
-      );
-    }
-    //CASE two named colors
-    else if (isItNamedColor(firstColor) && isItNamedColor(secondColor)) {
-      ratioResult.innerHTML = calculateRatio(
-        namesAndRGBValues[firstColor],
-        namesAndRGBValues[secondColor]
-      );
-    }
-    //CASE two HSL
-    else if (isValidHSL(firstColor) && isValidHSL(secondColor)) {
-      ratioResult.innerHTML = colorFormatRatio(
-        firstColor,
-        secondColor,
-        hslToRGB
-      );
-    } else {
-      showErrorMessage(warning);
-    }
+  if (isValidHSL(color)) {
+    return hslToRGB(color);
+  }
+
+  if (isValidRGB(color)) {
+    return rgbInputToRGBNumbers(color);
+  }
+
+  if (color in namesAndRGBValues) {
+    return namesAndRGBValues[color];
+  }
+  // invalid format or name
+  return null;
+};
+
+const displayRatioResult = () => {
+  // do nothing when any of the two inputs is empty
```

Comment:
I suggest that we do something, in this case.

Reply:
show info added `on blur`, when at least one input given